modules/sdk-coin-atom/src/lib/transaction.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  BaseKey,
+  BaseTransaction,
+  InvalidTransactionError,
+  ParseTransactionError,
+  PublicKey as BasePublicKey,
+  Signature,
+  TransactionRecipient,
+  TransactionType,
+} from '@bitgo/sdk-core';
+import { AtomTransaction, TransactionExplanation, TxData } from './iface';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { Buffer } from 'buffer';
+import { AtomTransactionType, sendMsgType } from './constants';
+import utils from './utils';
+import { encodePubkey, makeSignBytes } from '@cosmjs/proto-signing';
+import { encodeSecp256k1Pubkey } from '@cosmjs/amino';
+
+export class Transaction extends BaseTransaction {
+  private _atomTransaction: AtomTransaction;
+  private _signature: Signature;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  get atomTransaction(): AtomTransaction {
+    return this._atomTransaction;
+  }
+
+  setAtomTransaction(tx: AtomTransaction): void {
+    this._atomTransaction = tx;
+  }
+
+  /** @inheritDoc **/
+  get id(): string {
+    return this._id || 'UNAVAILABLE_TEXT';
+  }
+
+  addSignature(publicKey: BasePublicKey, signature: Buffer): void {
+    this._signatures.push(signature.toString('hex'));
+    this._signature = { publicKey, signature };
+    this.serialize();
+  }
+
+  get atomSignature(): Signature {
+    return this._signature;
+  }
+
+  /** @inheritdoc */
+  canSign(key: BaseKey): boolean {
+    return true;
+  }
+
+  /** @inheritdoc */
+  toBroadcastFormat(): string {
+    if (!this._atomTransaction) {
+      throw new InvalidTransactionError('Empty transaction');
+    }
+    return this.serialize();
+  }
+
+  /** @inheritdoc */
+  toJson(): TxData {
+    if (!this._atomTransaction) {
+      throw new ParseTransactionError('Empty transaction');
+    }
+    const tx = this._atomTransaction;
+    return {
+      id: this._id,
+      type: tx.type,
+      signerAddress: tx.signerAddress,
+      explicitSignerData: tx.explicitSignerData,
+      sendMessages: tx.sendMessages,
+      gasBudget: tx.gasBudget,
+    };
+  }
+
+  /** @inheritDoc */
+  explainTransaction(): TransactionExplanation {
+    const result = this.toJson();
+    const displayOrder = ['id', 'outputs', 'outputAmount', 'changeOutputs', 'changeAmount', 'fee', 'type'];
+    const outputs: TransactionRecipient[] = [];
+
+    const explanationResult: TransactionExplanation = {
+      displayOrder,
```

Comment:
Question: It does seem like this `TransactionExplanation` object is constructed in such a way that there is inherent ordering as per the `displayOrder` constant. Do we still need the constant for some other purpose ?

Reply:
I'm not entirely sure, we can ask the greater team and see if we can get more insight, but the constant is present for all other coins' implementation of this specific method.